cds-snc/notification-admin
# Banner for page previews

# Summary | Résumé

Adds a "preview" overlay for previewing pages from the CMS. The two parts of this are:
1. a 'phase-banner' type component (similar to [the "Alpha" banner in the header of Forms](https://forms-staging.cdssandbox.xyz/))
2. A full-screen yellow border 

The design goal here is that users should _always_ know they are previewing a page, not seeing live content, but the 'preview' interface itself shouldn't interfere with the intended layout of the content. (You don't want a 'preview' page that doesn't look how the published page will look.)

![Screen Recording 2022-01-19 at 09 58 30](https://user-images.githubusercontent.com/2454380/150157223-775282c7-2efa-4229-9be1-fe0b46aeac83.gif)

## Features

- Only on pages that use the "/preview" sub-path, not used anywhere else.
- Contains an "edit" link (Note that this will 404 for anyone not logged into GC Articles)
- English and French
- Updates the page title in the browser tab so that it's abundantly clear when it's a preview

(Note that this isn't actually working for "draft" content: currently we are using the `/preview` url to preview already-published content.)

## To test this

- See what it looks like on the "Features" page preview here: https://notification-preview-ba-wfkr8y.herokuapp.com/preview?id=11
- If you're logged into WordPress, you can edit the page here: https://articles.cdssandbox.xyz/notification-gc-notify/wp-admin/post.php?post=11&action=edit&lang=en
